### PR TITLE
feature/more-seattle-scraper-tests

### LINKED ISSUE #45: Seattle Scraper Unit Tests
### Feature Description

Create unit tests for the Seattle scraper.

### Use Case

Ensure that the Seattle scraper collects the data we need.

### Solution

Create PyTests that compare the actual values of each of the following features to their expected values:
- Number of events (legislative meetings) between two given dates
- Number of sessions in the first event
- The expected `caption_uri` for the first session of the first event
- The expected `video_uri` for the first session of the first event

### Alternatives

Keep the scrapers without writing unit tests.

### PR DESCRIPTION
**Pull request by Shak2000**
### Link to Relevant Issue

This pull request resolves #45 

### Description of Changes

I added new unit tests for the Seattle scraper.

### CODE REVIEW

**Comment by dphoria** [comment]
Thank you Shak. This will be very helpful moving forward. I have a number of thoughts. I'd love to hear how you feel about these points.

- We should use meetings with multiple sessions as answers. Testing the number of `EventIngestionModel.sessions` is good, but most have just 1 one, so we should pick one that has more than 1, e.g. from [10/27/21](https://seattle.legistar.com/MeetingDetail.aspx?ID=893278&GUID=D930A5A9-0548-411D-BCA5-131913878DFA&Options=info|&Search=).
- Checking the outermost item (`EventIngestionModel`) count isn't really testing much in my opinion. On top of testing the number of `EventIngestionModels`, I think we should also test the number of `EventIngestionModel.event_minutes_items` and the number of `EventMinutesItem.votes`, the really important data inside `EventIngestionModel`.

That's it for now; I'll post when more come to me.

Side note, I think the file name should use _ instead of -. I understand the test would be invoked on the command line using `pytest`, but if anyone does try to import the file, e.g. `from cdp_scrapers.tests.scraper-tests import whatever`, that line would fail because of the - (I believe).

**Comment by Shak2000** [comment]
> * We should use meetings with multiple sessions as answers. Testing the number of `EventIngestionModel.sessions` is good, but most have just 1 one, so we should pick one that has more than 1, e.g. from [10/27/21](https://seattle.legistar.com/MeetingDetail.aspx?ID=893278&GUID=D930A5A9-0548-411D-BCA5-131913878DFA&Options=info%7C&Search=).

It looks like each event can have only one session. You can see the list of events [here](https://seattle.legistar.com/Calendar.aspx). You can see that there is only one video, and therefore one session, for each event. For example, if you click on one of the events in the webpage the link above leads to, you would reach the details page of that event, such as [this one](https://seattle.legistar.com/MeetingDetail.aspx?ID=893277&GUID=978A2256-75D4-4764-B477-FB31C7025DC6&Options=info|&Search=), where you could find one video.

> * Checking the outermost item (`EventIngestionModel`) count isn't really testing much in my opinion. On top of testing the number of `EventIngestionModels`, I think we should also test the number of `EventIngestionModel.event_minutes_items` and the number of `EventMinutesItem.votes`, the really important data inside `EventIngestionModel`.

I will add tests for those attributes soon.

> Side note, I think the file name should use _ instead of -. I understand the test would be invoked on the command line using `pytest`, but if anyone does try to import the file, e.g. `from cdp_scrapers.tests.scraper-tests import whatever`, that line would fail because of the - (I believe).

Good catch! I fixed it.

**Comment by dphoria** [comment]
> It looks like each event can have only one session. You can see the list of events [here](https://seattle.legistar.com/Calendar.aspx). You can see that there is only one video, and therefore one session, for each event. For example, if you click on one of the events in the webpage the link above leads to, you would reach the details page of that event, such as [this one](https://seattle.legistar.com/MeetingDetail.aspx?ID=893277&GUID=978A2256-75D4-4764-B477-FB31C7025DC6&Options=info%7C&Search=), where you could find one video.
> 
Incorrect. Seattle can have meetings with multiple `sessions` and thus multiple videos. I chose the wrong date as an example. We specifically worked on the multiple session problem recently. Any meetings on the calendar with Video link and "session I ... session II" in the descriptions have multiple sessions//videos. For example the one on [10/26](https://seattle.legistar.com/MeetingDetail.aspx?ID=893277&GUID=978A2256-75D4-4764-B477-FB31C7025DC6&Options=info|&Search=) has sessions I and II; `SeattleScraper` also retrives 2 videos for it, correctly.

**Comment by dphoria** [comment]
I would recommend you run actually the scraper for the example you gave (https://seattle.legistar.com/MeetingDetail.aspx?ID=893277&GUID=978A2256-75D4-4764-B477-FB31C7025DC6&Options=info%7C&Search=) to convince yourself. Also, after you click on the video link, you have to scroll down to see there are videos for session I and II for that event, as it should.

**Comment by dphoria** [comment]
Shak I just ran the scraper for 10/26, and it did nto get all the videos. There is a bug I think. I will examine further later today. We just worked on recently. Sucks! LOL

File: cdp_scrapers/tests/scraper_tests.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from datetime import datetime
+
+
+@pytest.mark.parametrize(
+    "start_date_time, end_date_time, number_of_events,"
+    "number_of_sessions_in_first_event, number_of_event_minute_items,"
+    "event_with_votes, event_minute_item_with_votes, number_of_votes,"
+    "expected_caption_uri_in_first_session, expected_video_uri_in_first_session",
+    [
+        # Check for 1 event with 1 session and 10 event minute items.
```
Comment (dphoria) [inline cdp_scrapers/tests/scraper_tests.py:13]: To me, these first 2 tests don't gain us anything that the latter 2 do not.
This first one tests for multiple event minute items. Well, the last one tests for that as well, along with votes.
The second test checks getting multiple events from a period of time, which again the last test does.
So I feel like the first 2 tests are kind of meaningless. Am I missing something Shak? I feel like tests 3 and 4 test for everything the first 2 do, and then more, like number of sessions and votes.

Comment (Shak2000) [reply cdp_scrapers/tests/scraper_tests.py]: I removed the first 2 Seattle tests now.

**Comment by Shak2000** [comment]
Should we add Git Actions to run the unit scraper tests every time there is a Git push?

**Comment by dphoria** [comment]
That's a great idea Shak. I think, though, that can be a different PR; to keep PRs isolated as possible. But I am OK either way. What do you think?

For this PR, please make another commit after linting / formatting fix. You can run `make build` or `tox` to check locally. Or see the result on the Checks tab.

**Comment by Shak2000** [comment]
> That's a great idea Shak. I think, though, that can be a different PR; to keep PRs isolated as possible. But I am OK either way. What do you think?
> 
> For this PR, please make another commit after linting / formatting fix. You can run `make build` or `tox` to check locally. Or see the result on the Checks tab.

I agree that the Git Actions should be put into another PR. I will also look into how I can run `make build` or `tox` to check my repo locally.

**Comment by dphoria** [comment]
Thank you Shak! This will be useful.

**Pull request merged**